src/iuse_actor.cpp

```diff
@@ -973,27 +972,59 @@ bool firestarter_actor::prep_firestarter_use( const player &p, tripoint &pos )
     if( g->m.flammable_items_at( pos ) ||
         g->m.has_flag( "FLAMMABLE", pos ) || g->m.has_flag( "FLAMMABLE_ASH", pos ) ||
         g->m.get_field_strength( pos, fd_web ) > 0 ) {
-        // Check for a brazier.
-        bool has_unactivated_brazier = false;
+        
+        bool has_unactivated_brazier = false; // Check for a brazier.
+        bool has_tinder_here = false;
         for( const auto &i : g->m.i_at( pos ) ) {
             if( i.typeId() == "brazier" ) {
                 has_unactivated_brazier = true;
             }
+            if( i.is_tinder() ) {
+                has_tinder_here = true;
+            }
         }
         if( has_unactivated_brazier &&
             !query_yn(_("There's a brazier there but you haven't set it up to contain the fire. Continue?")) ) {
             return false;
         }
-        return true;
+
+        inventory &inv = g->u.inv;
+        item &tinder = inv.find_item( inv.position_by_type( "tinder" ) );
```

Comment:
As you're using `tinder` only inside one of the branches below, it should only be created there (as close to its usage as possible).